src/App.js

```diff
@@ -1,20 +1,38 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import TodoList  from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+const useSemiPersistentState = () => {
+
+  
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
If the local storage happened to be empty, you will have a problem with the value of `todoList` being `null`. It would be better if it defaults to an empty array since that is the starting value that would be best for an empty todo list. `JSON.parse(localStorage.getItem('savedTodoList')) || []
  );`